### PR TITLE
Refactor to common (work in progress)

### PR DESCRIPTION
**Pull request by shakeelmohamed**
**This is very much a work in progress, but I wanted to get the bulk of it started.**

When this is actually done, `js/zap-common.js` will be moved into a bower module and end up moving to something like: `bower_components/zap-common/index.js`

Looking for some feedback here @zen-audio-player/contributors :smiley:

### CODE REVIEW

File: js/zap-common.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    $.getJSON("https://www.googleapis.com/youtube/v3/search", {
+        key: youTubeDataApiKey,
+        part: "snippet",
+        q: query,
+        type: "video"
+    }, onData).fail(onFail);
+}
+
+function getAutocompleteSuggestions(query, callback) {
+    $.getJSON("https://suggestqueries.google.com/complete/search?callback=?", {
```
Comment (shakeelmohamed) [inline js/zap-common.js]: @monicacheung this was from bb3b87c87b5b751ff2738a0a87394d126128db48, why is `callback=?` in the URL? Does it have any effect?

**Comment by monicacheung** [comment]
@shakeelmohamed To use jsonp for cross-domain request. Without it, we'll see origin not allowed err.

**Comment by monicacheung** [comment]
On a side note, we can get rid of the file:// running site locally err if in the google dev console, you remove whatever's in "Accept requests from these HTTP referrers (web sites) (Optional)".

File: js/zap-common.js (hunk 1 above)
Comment (monicacheung) [inline js/zap-common.js]: @shakeelmohamed To use jsonp for cross-domain request. Without it, we'll see origin not allowed err.

File: js/zap-common.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        // We found a slash in the video ID (ex: real id is ABC123, but saw ABC123/zen)
+        // So, only keep what's before the slash
+        if (slashPos !== -1) {
+            videoInfo.id = videoInfo.id.substring(0, slashPos);
+        }
+
+        return videoInfo;
+    }
+}
+/*eslint-enable */
```
Comment (monicacheung) [inline js/zap-common.js]: getVideoDescription() should go in this file too.

Comment (shakeelmohamed) [inline js/zap-common.js]: done, although I'm not sure it will be used by another project

**Comment by shakeelmohamed** [comment]
>On a side note, we can get rid of the file:// running site locally err if in the google dev console, you remove whatever's in "Accept requests from these HTTP referrers (web sites) (Optional)".

@monicacheung Yeah, but the restrictions are there to prevent abuse. I can't justify removing them at this time, since it's only going to affect contributors.

**Pull request merged**